Firestore/core/src/core/sync_engine.cc

```diff
@@ -567,6 +573,74 @@ void SyncEngine::RemoveLimboTarget(const DocumentKey& key) {
   PumpEnqueuedLimboResolutions();
 }
 
+void SyncEngine::LoadBundle(std::shared_ptr<bundle::BundleReader> reader,
+                            std::shared_ptr<api::LoadBundleTask> result_task) {
+  auto bundle_metadata = reader->GetBundleMetadata();
+  if (!reader->reader_status().ok()) {
+    LOG_WARN("Failed to GetBundleMetadata() for bundle with error %s",
+             reader->reader_status().error_message());
+    result_task->SetError(reader->reader_status());
+    return;
+  }
+
+  auto has_newer_bundle = local_store_->HasNewerBundle(bundle_metadata);
+  if (has_newer_bundle) {
+    result_task->SetSuccess(SuccessProgress(bundle_metadata));
+    // TODO: reader.close()?
```

Comment:
Leftover TODO?

Reply:
I think we don't need to worry about manually closing.